java-frontend/src/main/java/org/sonar/java/xml/maven/PomCheckContextImpl.java

```diff
@@ -67,15 +68,28 @@ public void reportIssue(PomCheck check, int line, String message, List<Location>
   }
 
   @VisibleForTesting
-  static AnalyzerMessage getSecondaryAnalyzerMessage(JavaCheck check, File file, Location location) {
-    XmlLocation startLocation = location.tree.startLocation();
-    XmlLocation endLocation = location.tree.endLocation();
-    TextSpan ts = new TextSpan(
-      startLocation.line(), offsetFromColumn(startLocation.column()),
-      endLocation.line(), offsetFromColumn(endLocation.column()));
+  AnalyzerMessage getSecondaryAnalyzerMessage(JavaCheck check, File file, Location location) {
```

Comment:
Sorry, comment should have been one line below. You only need `TextSpan` in the location class. Up to you if this is too much to be there.